apps/showcase/src/components/showcase/forms/forms-pres.component.ts

```diff
@@ -0,0 +1,158 @@
+
+  /** The form control object bind to the personal info component */
+  public personalInfoFormControl: UntypedFormControl;
+  /** The form control object bind to the emergency contact component */
+  public emergencyContactFormControl: UntypedFormControl;
+
+  public submittedFormValue = '';
+
+  public firstSubmit = true;
+  public firstEmergencyContactFormSubmit = true;
+  public firstPersonalInfoFormSubmit = true;
+  private readonly forbiddenName = 'Test';
+
+  constructor() {
+    this.translations = translations;
+    this.personalInfo = { name: '', dateOfBirth: this.formatDate(Date.now()) };
+    this.personalInfoFormControl = new UntypedFormControl(this.personalInfo);
+    this.emergencyContact = { name: '', phone: '', email: '' };
+    this.emergencyContactFormControl = new UntypedFormControl(this.emergencyContact);
+    this.personalInfoValidators = {
+      global: formsPresValidatorGlobal(this.forbiddenName, translations.globalForbiddenName, translations.globalForbiddenNameLong, { name: this.forbiddenName }),
+      fields: {
+        dateOfBirth: dateCustomValidator(translations.dateInThePast)
+      }
+    };
+    this.emergencyContactValidators = {
+      global: formsPresValidatorGlobal(this.forbiddenName, translations.globalForbiddenName, translations.globalForbiddenNameLong, { name: this.forbiddenName })
+    };
+  }
+
+  private formatDate(dateTime: number) {
+    return formatDate(dateTime, 'yyyy-MM-dd', 'en-GB');
+  }
+
+  /** This will store the function to make the personal info form as dirty and touched */
+  public _markPersonalInfoInteraction: () => void = () => {};
+  /** This will store the function to make the emergency contact form as dirty and touched */
+  public _markEmergencyContactInteraction: () => void = () => {};
+
+  /**
+   * Register the function to be called to mark the personal info form as touched and dirty
+   *
+   * @param fn
+   */
+  public registerPersonalInfoInteraction(fn: () => void) {
+    this._markPersonalInfoInteraction = fn;
+  }
+
+  /**
+   * Register the function to be called to mark the personal emergency contact form as touched and dirty
+   *
+   * @param fn
+   */
+  public registerEmergencyContactInteraction(fn: () => void) {
+    this._markEmergencyContactInteraction = fn;
+  }
+
+  /** submit function */
+  public submitAction() {
+    if (this.firstSubmit) {
+      this._markPersonalInfoInteraction();
+      this._markEmergencyContactInteraction();
+      this.firstSubmit = false;
+    }
+    const isValid = !this.personalInfoFormControl.errors && !this.emergencyContactFormControl.errors;
+    if (isValid) {
+      this.submittedFormValue = JSON.stringify(this.personalInfoFormControl.value) + '\n' + JSON.stringify(this.emergencyContactFormControl.value);
+      // eslint-disable-next-line no-console
+      console.log('FORMS COMPONENT: personal info form status', this.personalInfoFormControl.status);
```

Comment:
You should probably leave the submitForm logic to each form and just call submitPersonalInfoForm and the other one.
This would highlight that you do not need to duplicate each blocks submit action logic but let them handle it.

Reply:
I would have prefer two different component to really highlight each component is the owner of its business logic but we can keep it this way if you think that would be overkill